### PR TITLE
Fix #173 regression on --token

### LINKED ISSUE #173: Token option is not working
apigeetool isn't working when using the -t option, to use the Apigee Access token. The code that was working previously, is failing with this error.

`Invalid arguments: Error: Missing required option with no prompt "username"`

This is my command that was working.
`apigeetool deployproxy  --token $TOKEN -o $(ORG) -e dev1 -n $PROXYNAME  -L https://api.enterprise.apigee.com -i `

### PR DESCRIPTION
**Pull request by kurtkanaskie**
Using the --token option turns off prompting for other parameters.
Given that behavior, a required param without a prompt setting is not an error condition.

$ npm test
> apigeetool@0.14.2 test 
> jasmine test/testoptions.js
Randomized with seed 67608
Started
.............
13 specs, 0 failures

### CODE REVIEW

**Comment by kurtkanaskie** [comment]
It's already in the README.
I updated the -h output:
`
$ apigeetool fetchProxy -h
api          -n (required) API Name             (command specific)
asynclimit   -a (optional) Async limit                            
baseuri      -L (optional) Base URI                               
cafile       -c (optional) CA file                                
certfile     -C (optional) Cert file                              
debug        -D (optional) Debug                                  
file         -f (optional) fileName             (command specific)
header       -H (optional) Header                                 
help         -h (optional) Help                                   
insecure     -k (optional) insecure                               
json         -j (optional) JSON                 (command specific)
keyfile      -K (optional) Key file                               
netrc        -N (optional) netrc                                  
organization -o (required) Organization                           
password     -p (required) Password or -t or -N                   
revision     -r (required) Revision             (command specific)
token        -t (optional) Token                                  
username     -u (required) Username or -t or -N                   
verbose      -V (optional) Verbose                                
`

**Comment by whitlockjc** [comment]
I get it.  I'm just saying the _"or -t or -N"_ could be done at the  `-t` or `-N` locations.  Here is an example:

```
$ apigeetool fetchProxy -h
api          -n (required) API Name             (command specific)
asynclimit   -a (optional) Async limit                            
baseuri      -L (optional) Base URI                               
cafile       -c (optional) CA file                                
certfile     -C (optional) Cert file                              
debug        -D (optional) Debug                                  
file         -f (optional) fileName             (command specific)
header       -H (optional) Header                                 
help         -h (optional) Help                                   
insecure     -k (optional) insecure                               
json         -j (optional) JSON                 (command specific)
keyfile      -K (optional) Key file                               
netrc        -N (optional) netrc (overrides -p/-u)                                  
organization -o (required) Organization                           
password     -p (required) Password                   
revision     -r (required) Revision             (command specific)
token        -t (optional) Token (overrides -p/-u)                                  
username     -u (required) Username                   
verbose      -V (optional) Verbose
```

**Comment by kurtkanaskie** [comment]
Yeah, I considered that too. I didn't want users to think username was required along with token.
I don't have a strong preference, your's looks good. 
Your call.

**Comment by whitlockjc** [comment]
I think having documentation on the options that override other options makes the most sense, my approach _(not that I care about credit, just using "my approach" to differentiate the two options)_.  I'd love to see a `\t` in the string between the option name and its extra details, so as to align with the `(command specific)` strings.  For example, `token\t(overrides -p/-u)` instead of ``token\ (overrides -p/-u)`.

If you find wording to be better than _"overrides -p/-u"_, feel free to use it.

**Comment by kurtkanaskie** [comment]
Good by me, thanks!
```
$ ./lib/cli.js fetchProxy -h
api          -n (required) API Name     (command specific)
asynclimit   -a (optional) Async limit                    
baseuri      -L (optional) Base URI                       
cafile       -c (optional) CA file                        
certfile     -C (optional) Cert file                      
debug        -D (optional) Debug                          
file         -f (optional) fileName     (command specific)
header       -H (optional) Header                         
help         -h (optional) Help                           
insecure     -k (optional) insecure                       
json         -j (optional) JSON         (command specific)
keyfile      -K (optional) Key file                       
netrc        -N (optional) netrc        (overrides -p/-u) 
organization -o (required) Organization                   
password     -p (required) Password                       
revision     -r (required) Revision     (command specific)
token        -t (optional) Token        (overrides -p/-u) 
username     -u (required) Username                       
verbose      -V (optional) Verbose                        
```

File: lib/defaults.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ var DefaultDescriptor = {
     prompt: false
   },
   netrc: {
-    name: 'netrc',
+    name: 'Netrc',
```
Comment (whitlockjc) [inline lib/defaults.js]: Minor nit, this change isn't related to what you're trying to accomplish so I'd rather it not be in this PR.

File: lib/defaults.js [hunk 2]
```diff
@@ -98,7 +100,7 @@ var DefaultDescriptor = {
     scope: 'default'
   },
   insecure: {
-    name: 'insecure',
+    name: 'Insecure',
```
Comment (whitlockjc) [inline lib/defaults.js]: Same as above, regarding being out of scope.

File: lib/defaults.js [hunk 3]
```diff
@@ -58,7 +58,8 @@ var DefaultDescriptor = {
     name: 'Organization',
     shortOption: 'o',
     scope: 'default',
-    required: true
+    required: true,
+    prompt: true
```
Comment (whitlockjc) [inline lib/defaults.js]: Same as above, regarding being out of scope.

File: lib/defaults.js [hunk 4]
```diff
@@ -80,6 +81,7 @@ var DefaultDescriptor = {
   json: {
     name: 'JSON',
     shortOption: 'j',
+    scope: 'default',
```
Comment (whitlockjc) [inline lib/defaults.js]: Same as above, regarding being out of scope.

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "apigeetool",
-  "version": "0.14.1",
+  "version": "0.14.2",
```
Comment (whitlockjc) [inline package.json]: Please don't change this, we change it at release time.

File: package-lock.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "apigeetool",
-  "version": "0.13.0",
+  "version": "0.14.2",
```
Comment (whitlockjc) [inline package-lock.json]: This should get bumped during release as well.

File: lib/defaults.js (hunk 1 above)
Comment (kurtkanaskie) [reply lib/defaults.js]: Done

**Comment by whitlockjc** [comment]
What happened to the `lib/defaults.js` changes?  Isn't that where you add the `\t(overrides -p/-u)`?

**Comment by kurtkanaskie** [comment]
It's in the options.js file, that's where the additional information (command specific) and (overrides -p/-u) get added and properly spaced.
https://github.com/apigee/apigeetool-node/pull/177/commits/34918b7b842105b1965216c7fea06df54ffcd014

**Comment by whitlockjc** [comment]
That's the problem, I don't see those changes in `lib/options.js` anymore.

**Comment by kurtkanaskie** [comment]
It should be in the commit labeled: "Adjusted help output for -t and -N overrides" line #218.
The one just before you requested changes.

**Comment by whitlockjc** [comment]
Ah, I see!

**Pull request merged**